sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CombineFnsTest.java

```diff
@@ -289,25 +291,29 @@ public String apply(String input) {
       }
     };
 
-    DisplayDataCombineFn combineFn1 = new DisplayDataCombineFn("combineFn1");
-    DisplayDataCombineFn combineFn2 = new DisplayDataCombineFn("combineFn2");
+    DisplayDataCombineFn combineFn1 = new DisplayDataCombineFn("value1");
+    DisplayDataCombineFn combineFn2 = new DisplayDataCombineFn("value2");
 
     CombineFns.ComposedCombineFn<String> composedCombine = CombineFns.compose()
         .with(extractFn, combineFn1, new TupleTag<String>())
         .with(extractFn, combineFn2, new TupleTag<String>());
 
     DisplayData displayData = DisplayData.from(composedCombine);
-    assertThat(displayData, includes(combineFn1));
-    assertThat(displayData, includes(combineFn2));
+    assertThat(displayData, hasDisplayItem("combineFn1", combineFn1.getClass()));
+    assertThat(displayData, hasDisplayItem("combineFn2", combineFn2.getClass()));
+
+    String nsBase = DisplayDataCombineFn.class.getName();
+    assertThat(displayData, includes(combineFn1, nsBase + "$1"));
+    assertThat(displayData, includes(combineFn2, nsBase + "$2"));
   }
 
   private static class DisplayDataCombineFn extends Combine.CombineFn<String, String, String> {
     private final String value;
-    private final String key;
     private static int i;
+    private final int id;
 
     DisplayDataCombineFn(String value) {
-      this.key = "key" + (++i);
+      id = ++i;
```

Comment:
I'd still suggest passing the `id` in. Its simpler to read the test since everything is declared there and it doesn't complicate this much.